my-app/src/context/GameContext.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, createContext, useContext } from "react";
+import calculateWinner from "../utils/util";
+export const GameContext = createContext();
+
+export function GameProvider({ children }) {
+    const [history, setHistory] = useState([Array(9).fill(null)]);
+    const [stepNumber, setStepNumber] = useState(0);
+    const [xIsNext, setXisNext] = useState(true);
+    const contextValue = {
+        history, setHistory, stepNumber, setStepNumber, xIsNext, setXisNext
+    };
+    return <GameContext.Provider value={contextValue}>{children}</GameContext.Provider>
+};
+
+export const useHandleClick = () => {
```

Comment:
It's not must but it would be good to rename this custom hook. This is too generic name. For example, `usePlayATurn`. It should express logical flow of the tic-tac-toe game.